src/HttpMethodsClient.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface for HTTP conventional methods
+ *
+ * Use this interface when you do not have PSR-7 RequestInterface instances available.
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface HttpMethodsClient
+{
+    /**
+     * Sends a GET request
+     *
+     * @param string|UriInterface $uri
+     * @param array               $headers
+     * @param array               $options
+     *
+     * @throws Exception
+     *
+     * @return ResponseInterface
+     */
+    public function get($uri, array $headers = [], array $options = []);
```

Comment:
are you aware that technically a GET request can also have a body? elasticsearch is doing this extensively for example.

not sure if we need to care here or just point people to the message factory for that use case. on the other hand having the body would make the method signatures consistent